File: 14.最长公共前缀.go
/*
 * @lc app=leetcode.cn id=14 lang=golang
 *
 * [14] 最长公共前缀
 */

// @lc code=start
func longestCommonPrefix(strs []string) string {
	longestPrefix := slices.Max(strs)

	for _, str := range strs {
		i, j := 0, 0
		for i < len(longestPrefix) && j < len(str) {
			if longestPrefix[i] != str[j] {
				break
			}

			i++
			j++
		}

		longestPrefix = longestPrefix[:i]
	}

	return longestPrefix
}

// @lc code=end

File: 14.最长公共前缀.py
#
# @lc app=leetcode.cn id=14 lang=python3
#
# [14] 最长公共前缀
#

# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        maxStr = max(strs)
        minStr = min(strs)

        for i in range (0, len(minStr)):
            if maxStr[i] != minStr[i]:
                return minStr[:i]
        return minStr
# @lc code=end

